cluster/gce/gci/configure-helper.sh

```diff
@@ -43,6 +43,39 @@ function setup-os-params {
   echo "core.%e.%p.%t" > /proc/sys/kernel/core_pattern
 }
 
+# secure_random generates a secure random string of bytes. This function accepts
+# a number of secure bytes desired and returns a base64 encoded string with at
+# least the requested entropy. Rather than directly reading from /dev/random, we
+# use uuidgen which calls getrandom(2) which verifies that the entropy pool has
+# been initialized sufficiently for the desired operation.
+#
+# ARGS:
+#   #1: number of secure bytes to generate. We round up to the nearest factor of 32.
+function secure_random {
+  local infobits="${1}"
+  if ((infobits <= 0)); then
+    echo "Requested insufficient entropy" 1>&2
+    return 1
+  fi
+
+  local out=""
+  while (( "${infobits}" > 0 )); do
+    # uuids have 122 random bits, sha256 sums have 256 bits, so concatenate
+    # three uuids and take their sum.
+    out+=$(
+     (
+       uuidgen --random;
+       uuidgen --random;
+       uuidgen --random;
+     ) | sha256sum \
+       | head -c 64 \
+       | xxd -r -p
+    );
```

Comment:
I ran this a few times, and sometimes get:
```
bash: warning: command substitution: ignored null byte in input
```
I think that might be losing some entropy?

Reply:
Good catch, this is fixed by defering the hex decode out of the loop to when we do the base64 encode.